src/components/main/compose/mod.rs

```diff
@@ -127,15 +125,31 @@ pub fn Compose(cx: Scope<Props>) -> Element {
                                 .map(|s| s.to_string())
                                 .collect::<Vec<_>>();
 
-                            if !text_as_vec.is_empty() {
-                                // clicking the send button is meaningless if there isn't a conversation.
-                                if let Some(id) = ext_conversation_id {
-                                    // TODO: We need to wire this message up to display differently
-                                    // until we confim whether it was successfully sent or failed
-                                    if let Err(_e) = warp::async_block_in_place_uncheck(rg.send(id, None, text_as_vec)) {
-                                        //TODO: Handle error
-                                    };
+                            if text_as_vec.is_empty() {
+                                return;
+                            }
+                            
+                            // clicking the send button is meaningless if there isn't a conversation. 
+                            if let Some(id) = current_chat {
+
+                                // mutate the state
+                                let cur = state.read().all_chats.get(&id).cloned();
+                                if let Some( mut conversation_info) = cur {
+                                    // for multiline messages, take at most 2 lines
+                                    let v: Vec<String> = text_as_vec.iter().take(2).cloned().collect();
+                                    let s: String = v.join("\n");
+                                    // the sizing of the conversation box is fixed, so approximate the needed string length using
+                                    // the placeholder text
+                                    let msg = s.chars().take(24).collect();
+                                    conversation_info.last_msg_sent = Some(LastMsgSent::new(msg));
+                                    state.write().dispatch(Actions::UpdateConversation(conversation_info));
```

Comment:
Maybe this should be done if sending is successful. Maybe something like (the suggestion is an example, no need to accept it here)

```suggestion
match warp::async_block_in_place_uncheck(rg.send(id, None, text_as_vec)) {
  Ok(_) => {
     if let Some( mut conversation_info) = cur {
            // for multiline messages, take at most 2 lines
            let v: Vec<String> = text_as_vec.iter().take(2).cloned().collect();
            let s: String = v.join("\n");
            // the sizing of the conversation box is fixed, so approximate the needed string length using
            // the placeholder text
            let msg = s.chars().take(24).collect();
            conversation_info.last_msg_sent = Some(LastMsgSent::new(msg));
            state.write().dispatch(Actions::UpdateConversation(conversation_info));
      }
  }
  Err(_e) =>  {} //TODO: Handle error
};
```